atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameGetter.java

```diff
@@ -67,6 +77,44 @@ public String getInternalShortOverflowTableName(ConnectionSupplier connectionSup
         return getPrefixedOverflowTableName(tableRef);
     }
 
+    public Set<TableReference> getTableReferencesFromShortTableNames(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+        return getTableReferencesFromShortTableNamesWithPrefix(connectionSupplier, shortTableNames, tablePrefix);
+    }
+
+    public Set<TableReference> getTableReferencesFromShortOverflowTableNames(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+        return getTableReferencesFromShortTableNamesWithPrefix(
+                connectionSupplier, shortTableNames, overflowTablePrefix);
+    }
+
+    private Set<TableReference> getTableReferencesFromShortTableNamesWithPrefix(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames, String tablePrefixToStrip)
+            throws TableMappingNotFoundException {
+        Set<String> longTableNames = getLongTableNamesFromTableNames(connectionSupplier, shortTableNames);
+        return longTableNames.stream()
+                .peek(tableName -> {
+                    if (!StringUtils.startsWithIgnoreCase(tableName, tablePrefixToStrip)) {
+                        throw new SafeIllegalArgumentException(
+                                "Long table name does not begin with prefix",
+                                SafeArg.of("tableName", tableName),
+                                SafeArg.of("prefix", tablePrefixToStrip));
+                    }
+                })
+                .map(tableName -> StringUtils.removeStartIgnoreCase(tableName, tablePrefixToStrip))
+                .map(TableReference::fromInternalTableName)
+                .collect(Collectors.toSet());
+    }
+
+    private Set<String> getLongTableNamesFromTableNames(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+        if (useTableMapping) {
+            return oracleTableNameUnmapper.getLongTableNamesFromMappingTable(connectionSupplier, shortTableNames);
+        } else {
+            return shortTableNames;
+        }
```

Comment:
I'd suggest standardising with how the other `generate...` methods are written - in particular, not having an `else`-clause. 

Note that this *is* different from what you've seen in AtlasDB-Proxy internally! But in Atlas the canonical form is not to use an `else` block here.